### DISCUSSION TITLE
Any example to reduce the image size by reduce the quality, not the the size?

### DISCUSSION THREAD

**Discussion by jdang67** [saucecontrol/PhotoSauce#101] [Q&A]
Hello Everyone, I need to reduce the size of the images. However, I have not found any example to reduce the image size like reducing the quality.

thanks

jdang

**Comment by saucecontrol** [comment]
While that seems like a simple ask, there are literally infinite ways to achieve a file size reduction, and all of those ways involve some kind of trade-off in image fidelity and/or resolution.  They are also all highly dependent on the source image properties and content.

The tools for this purpose that do exist simply operate on trial and error.  For example, run a loop encoding with lower and lower quality settings until your target output file size is reached.  But even then, some images will not fit.  e.g. you can't fit a gigapixel space telescope image into 1MB no matter how low you go on quality.

What most apps end up doing is setting a maximum resolution for stored images, assuming that by reducing the resolution, the size will be kept in a reasonable range.

Hope that helps!
